packages/buckets/src/api/index.ts

```diff
@@ -512,21 +536,33 @@ export async function bucketsPushPath(
   key: string,
   path: string,
   input: any,
-  opts?: { progress?: (num?: number) => void },
+  opts?: PushOptions,
   ctx?: ContextInterface,
 ) {
   return new Promise<PushPathResult>(async (resolve, reject) => {
-    // Only process the first  input if there are more than one
+    // Only process the first input if there are more than one
     const source: File | undefined = (await normaliseInput(input).next()).value
     const client = grpc.client<PushPathRequest, PushPathResponse, APIServicePushPath>(APIService.PushPath, {
       host: api.serviceHost,
       transport: api.rpcOptions.transport,
       debug: api.rpcOptions.debug,
     })
+    // Send a close event to the bucket api upon abort
+    if (opts?.signal !== undefined) {
+      opts.signal.addEventListener('abort', () => {
+        client.close()
+        return reject(AbortError)
+      })
+    }
     client.onMessage((message) => {
+      // Let's just make sure we haven't aborted this outside this function
```

Comment:
We also want to check each time we receive a new message from the remote, if we've aborted this locally. If yes, close the connection right away and reject.